### PR TITLE
cabana: regenerate car_fingerprint_to_dbc when DBCs change

### PR DESCRIPTION
**Pull request by deanlee**

### CODE REVIEW

File: tools/cabana/SConscript [hunk 1]
```diff
@@ -39,7 +39,9 @@ if GetOption('extras'):
...
 
 output_json_file = 'tools/cabana/dbc/car_fingerprint_to_dbc.json'
-generate_dbc = cabana_env.Command('#' + output_json_file,
-                                  ['dbc/generate_dbc_json.py'],
+
+os.system('git log -n1 --format=format:"%H" > .generate_dbc_dependencies')
+generate_dbc = cabana_env.Command(['#' + output_json_file],
+                                  ['dbc/generate_dbc_json.py', '.generate_dbc_dependencies'],
                                   "python3 tools/cabana/dbc/generate_dbc_json.py --out " + output_json_file)
 cabana_env.Depends(generate_dbc, ["#common", "#selfdrive/boardd", "#opendbc", "#cereal"])
```
Comment (sshane) [inline tools/cabana/SConscript:47]:
```suggestion
cabana_env.Depends(generate_dbc, ["#common", "#selfdrive/boardd", Glob("#opendbc/*"), "#cereal"])
```

What about something like this?

Comment (deanlee) [reply tools/cabana/SConscript:47]: sonsc will iterate through the contents of all files and generate hash values. the performance can be much slower than check the  latest commit

Comment (sshane) [reply tools/cabana/SConscript:47]: What about *.dbc? If I rename/add/remove DBC file it should probably update

File: tools/cabana/SConscript [hunk 2]
```diff
@@ -42,4 +42,4 @@ output_json_file = 'tools/cabana/dbc/car_fingerprint_to_dbc.json'
 generate_dbc = cabana_env.Command('#' + output_json_file,
                                   ['dbc/generate_dbc_json.py'],
                                   "python3 tools/cabana/dbc/generate_dbc_json.py --out " + output_json_file)
-cabana_env.Depends(generate_dbc, ["#common", "#selfdrive/boardd", "#opendbc", "#cereal"])
+cabana_env.Depends(generate_dbc, ["#common", "#selfdrive/boardd", Glob("#opendbc/*.dbc"), "#cereal"])
```
Comment (sshane) [inline tools/cabana/SConscript:45]: @deanlee Actually we need both. Jenkins is failing on the random build test

Comment (deanlee) [reply tools/cabana/SConscript:45]: @sshane what about this: https://github.com/commaai/openpilot/pull/29832

**Pull request merged**